msdk-featuredetection/msdk-targeteddetection/src/main/java/io/github/msdk/featuredetection/targeteddetection/TargetedDetectionMethod.java

```diff
@@ -164,14 +173,25 @@ public TargetedDetectionMethod(@Nonnull List<IonAnnotation> ionAnnotations,
                 chromatogram.setDataPoints(newDataPoints);
             }
 
+            // Set the m/z value for the chromatogram
+            double[] mzValues = buildingChromatogram.getMzValues();
+            double sum = 0;
+            int values = 0;
+            for(int i=0; i < buildingChromatogram.getSize() ; i++) {
+                if (mzValues[i] > 0) {
+                    values++;
+                    sum = sum + mzValues[i];
+                }
+            }
+            double newMz = sum/values;
+            chromatogram.setMz(newMz);
+
             // Add the chromatogram to the chromatogram list

```

Comment:
In MZmine 2 we used a median of the m/z values, so we should probably stick to that.
It might be a good idea to add a utility class that will calculate it for you, when given the chromatogram data points.